src/eterna/pose2D/Pose2D.ts

```diff
@@ -71,6 +73,24 @@ export default class Pose2D extends ContainerObject implements Updatable {
         super();
         this._poseField = poseField;
         this._editable = editable;
+
+        // kkk transfer 3D mouse move picking result to 2D Canvas and highlight/click corresponding base.
+        window.addEventListener('picking', (e) => {
+            const ce = <CustomEvent>e;
+            const closestIndex: number = ce.detail.resno;
+            if (ce.detail.action === 'hover') {
+                this.on3DPickingMouseMoved(closestIndex - 1);
+            } else if (ce.detail.action === 'clicked') {
+                if (Mol3DGate.scope/* && Mol3DGate.scope.threeView.metaState == 2 */) {
+                    this.simulateMousedownCallback(closestIndex - 1);
+                }
+            }
+        });
+    }
+
+    protected pEditMode: GameMode;
+    public setEditMode(editMode: GameMode) {
+        this.pEditMode = editMode;
```

Comment:
This is a significant code smell with respect to coupling. It seems that you're just using this to access Mol3DGate. Instead, what you should do is raise events that will be handled in the GameMode, specifically using constructs from the `signals` package in this repo (src/signals)